framework/include/base/MooseApp.h

```diff
@@ -742,9 +757,30 @@ class MooseApp : public ConsoleStreamInterface, public libMesh::ParallelObject
   void meshReinitForRMs();
 
   /**
-   * Function to check the integrity of the MeshMetaData data structure
+   * Function to check the integrity of the restartable meta data structure
    */
-  void checkMeshMetaDataIntegrity() const;
+  void checkMetaDataIntegrity() const;
+
+  ///@{
+  /**
+   * Iterator based access to the extra RestartableDataMap objects; see Checkpoint.C for use case.
+   *
+   * These are MOOSE internal functions and should not be used otherwise.
+   */
+  std::unordered_map<RestartableDataMapName,
```

Comment:
You might consider creating an alias for this type. It's pretty nasty.

Reply:
I will probably just make a struct with "data" and "suffix" members to make it clear on both ends.